src/q4/os-SINF1252/summary/os-SINF1252-summary.tex

```diff
@@ -952,7 +952,7 @@ \subsection{Les fonctions et procédures}
 La raison pour laquelle il faut à chaque fois qu'une procédure descende
 le sommet de la pile au début de son exécution et l'augmente à la
 fin est pour respecter une convention de l'architecture IA32 qui veut que les adresses de retour des procédures
-soient alignées sur des blocs de 16 bytes.
+soient alignées sur des blocs de 16 bits soit 4 bytes.
```

Comment:
Ca dépend des compilateurs mais en réalité c'est 16 bytes : https://en.wikipedia.org/wiki/X86_calling_conventions#cdecl

Reply:
Même en IA32? Ton article se réfère aux x86, je pense qu’il y a une grosse équivalence entre les deux, mais je ne sais pas si gcc change en conséquence... J’ai fais cette modif parce qu’en lisant j’avais trouvé bizarre que en tp, en assembleur, on monte et descend de 4 Bytes sur la pile (maintenant on manipule des int en tp) et que l’adresse de retour devait être de 16bytes

Reply:
On monte et descend de 4 bytes lorsqu'il s'agit d'allouer un entier sur la pile. Mais à l'échelle d'un appel à une procédure/fonction, l'alignement est de 16 bytes (=4 int) généralement, et c'est la convention qui est précisée dans le syllabus de sysinf (section "Les fonctions et procédures").
(By the way : 16 bits = 2 bytes en IA-32 ou en x86, pas 4)

Reply:
D’accord tu m’apprends un truc là. Moi qui pensais avoir bien lu le syllabus... J’ai été vérifier, et la phrase y est tel quel😅 (Oups pour le 16=>2... j’ai fait un raccourci un peu rapide)